vernac/mltop.ml

```diff
@@ -40,26 +40,36 @@ open Pp
 let coq_mlpath_copy = ref [Sys.getcwd ()]
 let keep_copy_mlpath path =
   let cpath = CUnix.canonical_path_name path in
-  let filter path' = not (String.equal cpath path')
-  in
+  let filter path' = not (String.equal cpath path') in
   coq_mlpath_copy := cpath :: List.filter filter !coq_mlpath_copy
 
-type plugin =
-| Legacy of { obj_file_path : string; fl_public_name : string }
-| Findlib of { fl_public_name : string }
+module PluginSpec = struct
+
+  type t = { file : string option; lib : string }
+
+  let encode { file; lib } = ( file, lib )
+  let decode ( file, lib ) = { file; lib }
```

Comment:
What's this for?

Reply:
To avoid having to mess with summary.ml too much, and still have a record. This is legacy code, so hopefully we can remove it at some point, when we drop support for Declare ML module `foo_plugin`.

Reply:
oh I see, on one side there's parentheses and a comma, on the other curly brackets and a semicolon.
I thought they were the identity when I first looked

Reply:
Yes, it is confusing, I could have used a pair in mltop too, I dunno.

Reply:
maybe write the types and add a comment
~~~ocaml
  (* we don't want to define the type in summary.ml so we have to send it an anonymous tuple *)
  let encode { file; lib } : _ * _ = ( file, lib )
  let decode ( file, lib ) : t = { file; lib }
~~~

Reply:
or we can do the commit I pushed with a bit of obj.magic in summary